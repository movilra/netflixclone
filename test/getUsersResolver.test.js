const {resolvers} =  require('../src');
const {getPrismaTestInstance} = require('./getPrismaTestInstance')

afterEach( async () => {
    await getPrismaTestInstance().mutation.deleteManyUsers({where:{}})
})


test('Query users all works', async () => {
    await getPrismaTestInstance().mutation.createUser(
        {
            data:{
                name:"tes1",
                lastname:"test",
                email:"test@gmail.com",
                password:"eltest",
                birth_date:"1992-08-09T00:00:00"
            }
        },`{ id }`);



    expect(
            await resolvers.Query.users(
                {},
                {},
                {db:getPrismaTestInstance()},
                `{name,lastname,email}`
            )
    ).toMatchSnapshot();


})


test('Query users all works', async () => {
   let user =  await getPrismaTestInstance().mutation.createUser(
        {
            data:{
                name:"tes1",
                lastname:"test",
                email:"test@gmail.com",
                password:"eltest",
                birth_date:"1992-08-09T00:00:00"
            }
        },`{ id }`);

    let token = jwt.sign({userId:user},APP_SECRET)


    expect(
            await resolvers.Query.users(
                {},
                {},
                {request:{Authorization:`JWT ${token}`}
                    db:getPrismaTestInstance()},
                `{name,lastname,email}`
            )
    ).toMatchSnapshot();


})