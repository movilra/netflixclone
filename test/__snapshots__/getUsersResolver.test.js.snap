// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query users all works 1`] = `
Array [
  Object {
    "email": "test@gmail.com",
    "lastname": "test",
    "name": "tes1",
  },
]
`;
